src/Arcus.Messaging.Abstractions/MessageHandling/MessageHandler.cs

```diff
@@ -127,7 +127,6 @@ private static ProcessMessageAsync DetermineMessageImplementation<TMessage, TMes
             return async (rawMessage, generalMessageContext, correlationInfo, cancellationToken) =>
             {
                 if (rawMessage is TMessage message
-                    && generalMessageContext.GetType() == typeof(TMessageContext)
```

Comment:
I removed this additional check because it's redundant (the check on L131 does the same) + this check caused the condition to evaluate to `false` when using the `AzureServiceBusSessionMessageContext`.
Although `AzureServiceBusSessionManagementMessageContext` is a `AzureServiceBusManagementMessageContext`, this comparison returned `false` because it was not exactly the same type.  (`IsAssignableFrom` would've been a better choice, but given the fact that we do a similar check on L131, I decided to just remove this check).

Reply:
If the suggestion in the other comment is applied, this change is not needed anymore, I believe.

Reply:
But why would you keep this additional check, if it's already covered by the check on L131 as well ?  (Ok, the check that is removed checks if the given context is the exact type, whereas the check on L131 will also evaluate to true when using a subclassed MessageContext.  (And I don't think that's an issue tbh)